modules/gdpr_fields/src/EntityTraversalContainerInjectionInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Drupal\gdpr_fields;
+
+use Drupal\Core\Entity\EntityInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a common interface for dependency container injection.
+ *
+ * This interface gives classes who need services a factory method for
+ * instantiation as well as the entity to be traversed.
+ */
+interface EntityTraversalContainerInjectionInterface {
+
+  /**
+   * Instantiates a new instance of this class.
```

Comment:
Can we reword this? Maybe "Creates an instance of the traversal for this specific entity" would be better.